docs/roman/associations/skycell_asn.rst

```diff
@@ -0,0 +1,91 @@
+
+.. code-block:: python
+
+		skycell_asn file_list -o my_program
+
+where the individual calibrated detector files are in the current directory and the -o is the root
+for the output associations. To get a complete list of options you can run the command with the
+\-h option
+
+.. code-block:: python
+
+		skycell_asn -h
+
+
+By knowing the structure of the conventional file name you can generate custom associations based
+on the information included in the file names.
+The current naming convention for the input files consists of a Visit_ID + Exposure_ID + detector + suffix
+Where the Visit_ID is a 19 digit string
+PPPPPCCAAASSSOOOVVV, with
+
+PPPPP = Program number
+CC = Execution plan number
+AAA = Pass number (within execution plan)
+SSS = Segment Number (within pass)
+OOO = Observation number (within the leg, not to be confused with the Observation ID)
+VVV = Visit Number (within the observation)
+
+and the Exposure_ID is a four digits designating the exposures
+
+eeee = Exposure number (within the visit)
+
+The detector is WFI01, WFI02, ... WFI18
+
+The suffix indicates the processing level of the file and to create products based on the
+skycells the suffix should generally be 'cal'.
+
+and the file name is constructed as,
+rPPPPPCCAAASSSOOOVVV_eeee_detector_suffix.asdf
+
+To give a more concrete example we'll use the APT example program for the
+High Latitude Wide Angle Survey (hlwas). The program is 00991, the execution plan number is 01,
+there are 12 passes, 5 segments, 4 observations and a various number of visits.
+For this example we want to select a single filter, say F158, and that is segment 003.
+So to generate the visit level associations for observation 001 we would select the files using the bash
+command line,
+
+.. code-block:: text
+
+		skycell_asn r0099101001003001001_*_cal.asdf  -o  r0099101001003001001 --product-type visit
+
+Where the wildcard selects all the exposures for visit 001 and generates associations based on the skycells
+the exposures will populate. This will generate associations based the skycells that the exposures can
+contribute data to. The association files will be json files with names based
+
+.. code-block:: text
+
+	r0099101001003001001_<skycell name>_<product_type>_<filter>_<release product name>_i2d_asn.json
+
+or for the selections above
+
+.. code-block:: text
+
+	r0099101001003001001_r257dp63x98y83_visit_F158_prompt_i2d_asn.json
+
+where the skycell name can vary based on the location on the celestial sphere and the i2d indicates
+that this is resampled 2-d imaging data. The release product name can be changed from the default
+by adding the optional argument --release-product <new name> to the command line.
+
+An analogous command to generate the pass level products, again setting segment to 003 to only select
+the F158 filter.
+
+.. code-block:: text
+
+		skycell_asn r0099101???003*_*_cal.asdf  -o  r0099101001003001001 --product-type pass
```

Comment:
I think the right example here is more like:
skycell_asn r0099101001???003*_*_cal.asdf -o r0099101001 --product-type pass
i.e., making the path specify both the execution plan and the pass (01001) and then using the three ??? to go over all segments before specifying observation 003 to get a particular filter.  Then the output root should include the activity / pass information but nothing else, and the script figures out the filter?

Reply:
good catch, I updated this from the visit and forgot to change that.